lib/Graphics/Image.js

```diff
@@ -261,55 +291,428 @@ class Image {
+
+		if (lastFittingLine == -1) return false
+
+
+
+		//totalHeight = lines[0].ascent + lines[lastFittingLine].descent + (lastFittingLine > 0 ? (lastFittingLine) * lineheight : 0)
+		totalHeight = (lastFittingLine+1) * lineheight
+		//console.log('total text height', totalHeight, 'bounding box height', h);
+
+		// since we are forcing the lineheight to 1.1, we have to calculate a new, smaller ascent and descent
+		let correctedAscent = Math.round(fontsize * 1.02)
+		let correctedDescent = lineheight - correctedAscent
+		correctedAscent = Math.round(lines[0].ascent)
+		correctedDescent = lineheight - correctedAscent
+
+		let xAnchor = x
+		switch (halign) {
+			case 'left':
+				this.context2d.textAlign = 'left'
+				xAnchor = x
+				break
+			case 'center':
+				this.context2d.textAlign = 'center'
+				xAnchor = (x + w) / 2
+				break
+			case 'right':
+				this.context2d.textAlign = 'right'
+				xAnchor = x + w
+				break
+		}
+
+		let yAnchor = 0
+		switch (valign) {
+			case 'top':
+				yAnchor = correctedAscent
+				break
+			case 'center':
+				yAnchor = Math.round((h - totalHeight) / 2 + correctedAscent)
+				break
+			case 'bottom':
+				yAnchor = h - totalHeight + correctedAscent
+				break
+		}
+		yAnchor += y
+		
+		this.context2d.fillStyle = color
+
+		for (let l = 0; l <= lastFittingLine; l += 1) {
+			const text = (lines[l].text.join(''))
+			//this.context2d.fillText(text, xAnchor, yAnchor)
+			if (fontsize < 10) { 
+				let textPath = this.context2d.outlineText(text) // trick for disabling Antialiasing, but it also breaks support for some unicode
+  			this.context2d.fill(textPath.offset(xAnchor, yAnchor))
+			} else {
+				this.context2d.fillText(text, xAnchor, yAnchor)
+			}
+			//this.horizontalLine(yAnchor - fontsize, 'rgb(255,0,255)')
+			//this.horizontalLine(yAnchor + correctedDescent, 'rgb(0, 255, 0)')
+			//this.horizontalLine(yAnchor - correctedAscent, ''''rgb(0,0,255)')
+			//this.horizontalLine(yAnchor, 'rgb(255, 0, 0)')
+			//console.log('Fontsize', fontsize, 'Lineheight', lineheight, 'asc', correctedAscent, 'des', correctedDescent, 'a+d', correctedAscent+correctedDescent);
+			
+			yAnchor += lineheight
+		}
+		
+		this.lastUpdate = Date.now()
+		return true
 	}
 
-	/*
-	Draws aligned text in an boxed area.
-	int x: bounding box top left horizontal value
-	int y: bounding box top left vertical value
-	int w: bounding box width
-	int h: bounding box height
-	string text: the text to drawBank
-	rgb-array color: color of the text
-	string fontindex: index of font, either 'icon' or something else
-	int spacing: how much space should be between letters, leave undefined for spacing of font
-	int size: font size multiplier
-	string halign: horizontal alignment left, center, right
-	string valign: vertical alignment top, center, bottom
-	bool dummy: don't actually draw anything if true, just return if the text fits
-
-	returns true if text fits
-*/
-	drawAlignedText(
+	drawAlignedText_old(
```

Comment:
it would be good to remove these old methods, and the large blob of font.js that this depends upon

Reply:
Yes, like I wrote, cleanup is to be done.